fixql-core/src/main/scala/com/iterable/graphql/compiler/QueryReducer.scala

```diff
@@ -0,0 +1,139 @@
+package com.iterable.graphql.compiler
+
+import com.iterable.graphql.Field
+import play.api.libs.json.JsArray
+import play.api.libs.json.JsObject
+import play.api.libs.json.JsValue
+import play.api.libs.json.Json
+import play.api.libs.json.Writes
+import slick.dbio.DBIO
+
+import scala.concurrent.ExecutionContext
+
+object QueryReducer {
```

Comment:
Elsewhere you'll see the main input that the developer needs to provide are `QueryMappings` where we have `type QueryMappings` is roughly defined as `type QueryMappings = Field.Annotated[FieldTypeInfo] => Field[Resolver] => Resolver`. The mappings yield you a resolver for each field in the schema. And the resolver can depend on the field's sub-resolvers.

But we actually use `case class QueryReducer[A](f: Field[Resolver[A]] => Resolver[A])` to wrap the resolver-producing function. So we actually have `type QueryMappings = Field.Annotated[FieldTypeInfo] => QueryReducer`.